cmd/run.go

```diff
@@ -369,6 +339,10 @@ a commandline interface for interacting with it.`,
 			logger.Warn("No data generated, because no script iterations finished, consider making the test duration longer")
 		}
 
+		if !conf.NoUsageReport.Bool {
+			_ = reportUsage(execScheduler)
```

Comment:
This is not done in a goroutine anymore because it's done after the end of execution, and unless we manually synchronize it, the process will exit before the report is submitted.

So currently this adds a slight delay before the summary report is shown, which might be annoying, but then there's always `K6_NO_USAGE_REPORT`.

Also, I changed from using `panic()` to returning an error, but since it's never checked, maybe we shouldn't bother? The `panic` doesn't seem right for this, though. Let me know.

Reply:
Could the summary take long enough time for there to be time for this to finish? 
I do know that on especially metric heavy tests it does take some time to show the end summary so maybe it is slow enough in which case I would argue simple sync.WaitGroup or chan struct{} will probably be simple enough and beneficial enough to offset the added code, otherwise -  no.
Can you run a couple of tests and see if it gets slower with the length of test?

I do agree that panicking is not a good idea but maybe log it as info?  or debug?

Reply:
> Can you run a couple of tests and see if it gets slower with the length of test?

Yes, you're right. If the test is long or metric-heavy then the time to process the summary is enough for the usage report to be submitted.

But we can't rely on that, which would make the usage results even less useful.

So I added a `WaitGroup` in 03d027d0. There's still a slight delay at the end, but it doesn't impact the end summary, and it can always be disabled with the env var.

> I do agree that panicking is not a good idea but maybe log it as info? or debug?

We shouldn't log or output anything at this point since it could mangle the summary report, or be printed below it, which is a bit unsightly.

I left it as returning an error that is not checked. We could submit it to Sentry just to catch it failing, but then that would be another request to make, so not sure.

Reply:
No need to log or submit any failures. This is not critical, just an optional best-effort ping.